src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -203,70 +293,44 @@
    * 
    * @return a {@link MergedByteBuffers} of the current read data for this client.
    */
-  public abstract MergedByteBuffers getRead();
+  public MergedByteBuffers getRead() {
+    MergedByteBuffers mbb = null;
+    synchronized(readerLock) {
+      mbb = readBuffers.duplicateAndClean();
+    }
+    if(mbb.remaining() >= maxBufferSize) {
+      se.setClientOperations(this);
+    }
+    return mbb;
+  }
+
   
   /**
    * 
-   * <p>Adds a {@link ByteBuffer} to the Clients readBuffer.  This is normally only used by the {@link SocketExecuterInterface},
+   * <p>Adds a {@link ByteBuffer} to the Clients readBuffer.  This is normally only used by the {@link SocketExecuter},
    * though it could be used to artificially inject data through the client.  Calling this will schedule a
    * calling the currently set Reader on the client.</p>
    * 
    * 
    * @param bb the {@link ByteBuffer} to add to the clients readBuffer.
    */
-  protected abstract void addReadBuffer(ByteBuffer bb);
-  
-  /**
-   * <p> This tries to write the ByteBuffer passed to it to the Client.  If the Clients writeBuffer 
-   * is greater than the maxbufferSize the {@link ByteBuffer} will not be added and false will be returned.
-   * If the current writeBuffer is less than the maxBufferSize then the ByteBuffer will be added and this
-   * will return true.</p>
-   * 
-   * <p>As this write is thread safe it should be noted that calling this with multiThreads can not garentee order.
-   * As such it is recommended to write full parsable protocol packets at a time, or implement your own locking around the
-   * write if you are streaming data raw data and using more then 1 thread to do such.
-   * </p>
-   * 
-   * @param bb the {@link ByteBuffer} to write to the client.
-   * @return true if the client has taken the ByteBuffer false if it did not.
-   */
-  public abstract boolean writeTry(ByteBuffer bb);
+  protected void addReadBuffer(final ByteBuffer bb) {
+    addReadStats(bb.remaining());
+    synchronized(readerLock) {
+      final int start = readBuffers.remaining();
+      readBuffers.add(bb);
+      if(readBuffers.remaining() > 0 && start == 0){
+        readerListener.call().onRead(this);
```

Comment:
since all the listeners have their own executor it would only be an advantage with a very high listener count (which I assume is unlikely). So probably not a big deal